.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,188 @@
+name: "Continuous Integration"
+
+on:
+  pull_request:
+    paths-ignore:
+      - 'mkdocs.yml'
+      - '**.md'
+  push:
+    branches:
+      - '[0-9]+.[0-9]+.x'
+      - 'refs/pull/*'
+    paths-ignore:
+      - 'mkdocs.yml'
+      - '**.md'
+
+jobs:
+  lint-composer-json:
+    name: Lint composer.json
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: json-syntax-check
+        run: jq empty composer.json
+
+  supported-checks-matrix:
+    name: Determine which checks to run
+    runs-on: ubuntu-latest
+    needs: [lint-composer-json]
+    outputs:
+      check: ${{ steps.supported-checks-matrix.outputs.check }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: supported-checks-matrix
+        run: |
+          # Update the tools def to add or exclude checks
+          CHECKS=$(cat composer.json | jq -c 'def tools:["benchmark","cs-check","static-analysis","test"] ; def intersect(x;y): (x + y | sort) as $sorted | reduce range(1; $sorted|length) as $i ([]; if $sorted[$i] == $sorted[$i-1] then . + [$sorted[$i]] else . end) ; .scripts | keys | intersect(.;tools)')
+          echo "Discovered checks: ${CHECKS}"
+          echo "::set-output name=check::${CHECKS}"
+
+  supported-versions-matrix:
+    name: Supported Versions Matrix
+    runs-on: ubuntu-latest
+    needs: [lint-composer-json]
+    outputs:
+      version: ${{ steps.supported-versions-matrix.outputs.version }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: supported-versions-matrix
+        uses: WyriHaximus/github-action-composer-php-versions-in-range@v1
+
+  dependency-versions-matrix:
+    name: Dependencies to Test Matrix
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.dependency-versions-matrix.outputs.check }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: dependency-versions-matrix
+        run: |
+          DEPS=$(if [ -f composer.lock ]; then echo -n '["lowest", "locked", "latest"]'; else echo -n '["lowest", "latest"]'; fi)
+          echo "Testing against dependencies: ${DEPS}"
+          echo "::set-output name=check::${DEPS}"
+
+  latest-supported-php-7:
+    name: Determine Latest Supported PHP 7 Version
+    runs-on: ubuntu-latest
+    needs: [supported-versions-matrix]
+    outputs:
+      version: ${{ steps.latest-supported-php-7.outputs.version }}
+    steps:
+      - id: latest-supported-php-7
+        run: |
+          VERSION=$(echo '${{ needs.supported-versions-matrix.outputs.version }}' | jq -r '. | map(. | select(startswith("7."))) | max')
+          echo "Latest PHP 7 version discovered: ${VERSION}"
+          echo "::set-output name=version::${VERSION}"
```

Comment:
I'm wondering if there might be a way to choose latest supported PHP **or** latest PHP 7, again, by flagging in the `composer.json` or elsewhere.

Reply:
From https://stackoverflow.com/a/32137942/2314626

there already exists such data:
- https://www.php.net/releases/?json=1&version=7

```
$ curl -s 'https://www.php.net/releases/?json=1&version=7' | jq -r .version
7.4.14
```

but you may want to keep the json in your organization to have more control on it

```sh
curl raw.github.com/laminas/.github/php-latest.json
curl raw.github.com/laminas/.github/php-latest-7.x.json
```

Reply:
This isn't about the latest PHP that php.net releases, but the latest PHP _the package supports_. But we can hard-code this to 7.4.

My current thought is to have a config file that this workflow consumes to determine if it should use latest minor release the package supports, or use the default (7.4).